### PR TITLE
Make sure that videos load correctly from different domains

### LINKED ISSUE #211: Videos from Youtube don't play in Min-Vid player after playing videos in the mini player from Vimeo or Vine
[Notes]:
- If videos from Youtube are played first, then Vimeo or Vine videos will not load.
- Attached a screen recording of the issue http://goo.gl/o30smt.

[Affected versions]:
- Firefox 45.0 and up
- Min-Vid 0.1.6-dev

[Affected Platforms]:
- All Windows
- All Mac 
- All Linux

[Prerequisites]:
- Latest Min-Vid add-on version (0.1.6-dev) was previously installed on a clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites. 
2. Navigate to a Vimeo or Vine
3. Send a video to the mini player and chose to play it.
4. Navigate to YouTube and send a video to the mini player.

[Expected result]:
- The selected video is played using the Min-Vid player.

[Actual result]:
- The video is sent to the player, but it doesn't load.

### PR DESCRIPTION
**Pull request by meandavejustice**
- fixes #211

### CODE REVIEW

File: components/player-view.js [hunk 1]
```diff
@@ -58,10 +58,17 @@ module.exports = React.createClass({
 
     requestAnimationFrame(this.step);
   },
-  componentWillMount: function() {
+  componentWillUpdate: function() {
     this.isYt = !!~this.props.domain.indexOf('youtube.com');
   },
-  componentDidMount: function() {
+  componentDidUpdate: function(prevProps, prevState) {
+    if (prevProps.src !== this.props.src) {
```
Comment (6a68) [inline components/player-view.js]: Will this line try to `attachVideoListeners` if src is reset to an empty string?

**Comment by 6a68** [comment]
this looks good but I'm out of time and still need to test. I'll finish looking tomorrow morning :beers:

**Comment by 6a68** [comment]
@meandavejustice This PR doesn't fix #211 for me on today's nightly:

- go to vimeo.com
- send any video to the player
- hit play
- go to youtube.com
- send any video to the player
- player is not updated

**Comment by meandavejustice** [comment]
@6a68 There was a case where we were calling `ytCtrl.getTime` prematurely. It's fixed up now.

**Comment by 6a68** [comment]
works for me, great job :+1:

**Pull request merged**